R-package/tests/testthat/test_lgb.unloader.R

```diff
@@ -2,7 +2,13 @@ VERBOSITY <- as.integer(
     Sys.getenv("LIGHTGBM_TEST_VERBOSITY", "-1")
 )
 
+CALCULATING_TEST_COVERAGE <- Sys.getenv("LIGHTGBM_TEST_COVERAGE") == "true"
```

Comment:
Can we use built-in `R_COVR` env variable instead?
https://www.rdocumentation.org/packages/covr/versions/3.5.1/topics/in_covr

Reply:
oh nice. I like `covr::in_covr()` a lot better than this solution with environment variables, didn't know about that! Thank you!

updated in https://github.com/microsoft/LightGBM/pull/4922/commits/1e1e491d6af3abbd01786309b3e99bf0e9255cc5

Reply:
Great! Happy to help with some simplification.
I see you omitted `if (require(covr))` part. What will happen if user doesn't have `covr` package installed but decided to run all tests with `testthat`?

Reply:
I guess failing tests illustrate that there will be an error in that case:

```
  Error in `loadNamespace(x)`: there is no package called 'covr'
```

```
* checking for unstated dependencies in ‘tests’ ...  WARNING
'::' or ':::' import not declared from: ‘covr’
* checking tests ...
  Running ‘testthat.R’ [26s/14s]
    ERROR
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
   6. base:::doWithOneRestart(return(expr), restart)
  
  ── 2. Error (test_lgb.unloader.R:31:5): lgb.unloader finds all boosters and remo
  <packageNotFoundError/error/condition>
  Error in `loadNamespace(x)`: there is no package called 'covr'
  Backtrace:
   1. testthat::skip_if(condition = covr::in_covr(), message = "lgb.unloader() tests are skipped when calculating test coverage") test_lgb.unloader.R:31:4
   3. base::loadNamespace(x)
   4. base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
   5. base:::withOneRestart(expr, restarts[[1L]])
   6. base:::doWithOneRestart(return(expr), restart)
  
  ══ DONE ════════════════════════════════════════════════════════════════════════
  Error: Test failures
  Execution halted
```

Reply:
haha yeah I tried to update that too fast, you're right.

Reply:
fixed in https://github.com/microsoft/LightGBM/pull/4922/commits/41e7f07566d110c86c7ddb7ff047cbc22258a298

had to include `# nolint` comments on those lines since we use `{lintr}` to discourage other uses of `require()`

https://github.com/microsoft/LightGBM/blob/af5b40e1f653e25a5f3470e09de4d1c68dc59663/.ci/lint_r_code.R#L71-L74

Reply:
just pushed one more commit adding `{covr}` to the `Suggests` dependencies, since `R CMD check` failed with the following:

```text
* checking for unstated dependencies in ‘tests’ ... WARNING
'::' or ':::' import not declared from: ‘covr’
'library' or 'require' call not declared from: ‘covr’
```

Reply:
gah I forgot that adding a new `Suggests` dependency would mean it has to be installed in CI

```text
* checking package namespace information ... OK
* checking package dependencies ... ERROR
Package suggested but not available: ‘covr’
```

https://github.com/microsoft/LightGBM/runs/4697717553?check_suite_focus=true

I don't think this change to support computing code coverage during development is worth adding a new dependency to CI jobs (which will make them take longer and increase the risk of failures).

I just pushed https://github.com/microsoft/LightGBM/pull/4922/commits/b97d37a3189b96ff2fe9b277cceda6c8e03eef2d, which proposes using the `R_COVR` environment variable.